sql/src/main/java/io/crate/operation/projectors/ShardDMLExecutor.java

```diff
@@ -171,13 +157,15 @@ private String getLocalNodeId(ClusterService clusterService) {
         return nodeId;
     }
 
-    private void processShardResponse(ShardResponse shardResponse) {
-        if (shardResponse.failure() != null) {
-            executionFuture.completeExceptionally(shardResponse.failure());
-            return;
+    private BitSet processShardResponse(ShardResponse shardResponse) {
+        Exception failure = shardResponse.failure();
+        if (failure != null) {
+            Throwables.throwIfUnchecked(failure);
+            throw new RuntimeException(failure);
```

Comment:
This is a bit related to the next change that I want to do - that is to encapsulate the result future in the `BatchIteratorBackpressureExecutor`. Imho it is otherwise very hard to reason about how it gets completed if it's completed from two different locations.

This `Throwables.throwIfUnchecked(failure); throw new RuntimeException(failure)` pattern is the replacement for the deprecated `Throwables.propagate`